configure.py

```diff
@@ -2402,5 +2403,54 @@ def create_build_system(args):
     generate_compdb('compile_commands.json', ninja, args.buildfile, selected_modes)
 
 
+def configure_using_cmake(args):
+    # all supported build modes, and if they are built by default if selected
+    build_by_default = {'debug': True,
+                        'release': True,
+                        'dev': True,
+                        'sanitize': False,
+                        'coverage': False}
+    selected_modes = args.selected_modes or build_by_default.keys()
+    selected_configs = ';'.join(mode.capitalize() for mode in selected_modes)
+    default_configs = ';'.join(mode.capitalize() for mode in selected_modes
+                               if build_by_default[mode])
+
+    settings = {
+        'CMAKE_CONFIGURATION_TYPES': selected_configs,
+        'CMAKE_CROSS_CONFIGS': default_configs,
+        'CMAKE_DEFAULT_CONFIGS': default_configs,
+        'CMAKE_CXX_COMPILER': args.cxx,
```

Comment:
What about CVMAKE_CXX_FLAGS?

Reply:
indeed. should have used `CMAKE_CXX_FLAGS` instead of `CMAKE_C_FLAGS`.